### PR TITLE
State Synchronization

### LINKED ISSUE #46: State Synchronization
Synchronize ship state with host periodically

### PR DESCRIPTION
**Pull request by mt-xing**
### Summary 

Periodically (every second) synchronize the state of all the doors and breaches against the definitive state provided by the host.
Close #46 

### Testing 

Tested on Windows by manually disabling synchronization of breach creation, breach resolution, and door creation. After 1 second, state sync worked as expected.

### Review Focus 

All interesting changes are in `mib`

### CODE REVIEW

File: source/MagicInternetBox.cpp [hunk 1]
```diff
@@ -121,6 +123,87 @@ void MagicInternetBox::sendData(NetworkDataType type, float angle, int id, int d
...
+			if (abs(doors[i]->getAngle() - angle) > FLOAT_EPSILON) {
+				CULog("Found open door that should be closed, id %d", i);
+				state->createDoor(angle, (int)i);
+			}
+		} else {
+			if (doors[i]->getAngle() != -1.0f) {
+				CULog("Found closed door that should be open, id %d", i);
+				state->flagDoor((int)i, 0, 1);
+				state->flagDoor((int)i, 1, 1);
+			}
```
Comment (mt-xing) [inline source/MagicInternetBox.cpp]: @ssorenson130 Can you review this part in particular? Is there a better way to forcefully open a door that was closed?

Comment (ssorenson130) [reply source/MagicInternetBox.cpp]: yes, this won't even really work for more than 2 players. just call `clear()` on the door

Comment (mt-xing) [reply source/MagicInternetBox.cpp]: The `clear()` method doesn't set the angle to `-1`

Comment (ssorenson130) [reply source/MagicInternetBox.cpp]: ah I misread this. what you added works fine

**Pull request merged**